app/models/qa/iri_template/variable_map.rb

```diff
@@ -13,23 +13,23 @@ class VariableMap
       attr_reader :default
       private :default
 
-      # @param [Hash] map configuration hash for the variable map
-      # @option map [String] :variable (required) name of the variable in the template (e.g. {?query} has the name 'query')
-      # @option map [String] :property (optional) always "hydra:freetextQuery" # TODO what other values are supported and what do they mean
-      # @option map [Boolean] :required (required) is this variable required
-      # @option map [String] :default (optional) value to use if a value is not provided in the request (default: '')
-      # @option map [Boolean] :encode (optional) whether to url_encode the value (default: false)
-      def initialize(map)
-        @variable = extract_variable(config: map)
-        @required = extract_required(config: map)
-        @default = extract_default(config: map)
-        @encode = extract_encode(config: map)
-        @property = map.fetch(:property, 'hydra:freetextQuery')
+      # @param [Hash] variable_map configuration hash for the variable map
+      # @option variable_map [String] :variable (required) name of the variable in the template (e.g. {?query} has the name 'query')
+      # @option variable_map [String] :property (optional) always "hydra:freetextQuery" # TODO what other values are supported and what do they mean
+      # @option variable_map [Boolean] :required (required) is this variable required
+      # @option variable_map [String] :default (optional) value to use if a value is not provided in the request (default: '')
+      # @option variable_map [Boolean] :encode (optional) whether to url_encode the value (default: false)
+      def initialize(variable_map)
+        @variable = Qa::LinkedData::Config::Helper.fetch_required(variable_map, :variable, nil)
+        @required = Qa::LinkedData::Config::Helper.fetch_boolean(variable_map, :required, nil)
+        @default = variable_map.fetch(:default, '').to_s
```

Comment:
@elrayle No need for changes, but I am noticing an inconstancy here.  In Some cases you use the Helper class and in other cases you access the map directly.  It might be better in the long run to be consistent.

Reply:
This makes sense.  I'll make that change.